File: Java/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<Character> stk = new Stack<>();
        for (char p:s.toCharArray()){
            if(p == '('){
                stk.push(p);
            }
            else{
                if(!stk.isEmpty()) { stk.pop(); }
                else { return false; }
            }
        }
                
        return stk.isEmpty();
    }
}

File: Python3/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    cnt = 0
    
    # 첫 번째에 )면 false
    if s[0] == ')':
        return False
    
    for i in range(len(s)):
        if s[i] == '(':
            cnt+=1
        else:
            cnt-=1
        
        if cnt < 0:
            return False
    return cnt == 0
